### DISCUSSION TITLE
reload secrets

### DISCUSSION THREAD

**Discussion by hmoffatt** [buildbot/buildbot#6588] [Q&A]
I've configured some secrets in a file using the SecretInAFile provider, but if the secrets change on disk while buildbot (3.4.0) is running they don't ever seem to be reloaded. Even a reload via signal does not update them, I had to restart the master. Is this expected?

**Comment by tardyp** [comment]
Indeed, secrets are not supposed to change often, so we have in memory caching that prevents reloading to work.
For vault, there is no caching, but for SecretInFile, if you don't change the file name, it won't get reloaded.

You may try to patch here
https://github.com/buildbot/buildbot/blob/e8d8fac1050428351d2a8ed145f12594154b251f/master/buildbot/secrets/providers/file.py#L77

and add following line in the end of that function to force a reconfiguration of the object each time (so a reload).

`self.configured = False`

**Comment by hmoffatt** [comment]
Thanks for your reply. I notice that this means even new secrets are not loaded on a reload. 

It is not really a problem, just a surprise during builder development when using test secrets initially.
